docs/guides/server/logging.adoc

```diff
@@ -71,15 +74,90 @@ To enable log handlers, enter the following command:
 
 <@kc.start parameters="--log=\"<handler1>,<handler2>\""/>
 
-The available handlers are
-<@profile.ifCommunity>
-`console`, `file`, `syslog`.
-</@profile.ifCommunity>
-<@profile.ifProduct>
-`console`, `file` and `syslog`.
-</@profile.ifProduct>
+The available handlers are:
+
+<@opts.expectedValues option="log"/>
+
 The more specific handler configuration mentioned below will only take effect when the handler is added to this comma-separated list.
 
+== Specify log level for each handler
+
+The `log-level` property specifies the global root log level and levels for selected categories.
+However, a more fine-grained approach for log levels is necessary to comply with the modern application requirements.
+
+To set log levels for particular handlers, properties in format `log-<handler>-level` (where `<handler>` is available log handler) were introduced.
+
+It means properties for log level settings look like this:
+
+* `log-console-level` - Console log handler
+* `log-file-level` - File log handler
+* `log-syslog-level` - Syslog log handler
+
+NOTE: The `log-<handler>-level` properties are available only when the particular log handlers are enabled.
+More information in log handlers settings below.
+
+Accepted log levels are specified in <<Log levels>> section, and *must be in lowercase*.
+
+=== General principle
+
+It is necessary to understand that setting the log levels for each particular handler *does not override the root level* specified in the `log-level` property.
+It specifies the minimal allowed level for logs shown in the output.
+It respects levels specified in the `log-level` property, which creates a minimal log level bar for log records.
+
+Specifically, when an arbitrary log level is specified for the handler, it does not mean the log records with the log level will be present in the output.
+The root log level needs to be assessed as well, as it determines the minimal level for exposed log records.
+
+Log handler levels provide the *restriction for the root log level*, so the default value is `all` - without any restriction.
+
+==== Visualization
```

Comment:
I was thinking if we even need the visualization. It'd be a bit hard to maintain if changes are needed (without source files). And the examples (if we move the written explanation from here to there) are good on its own, IMHO. @mabartos WDYT?

Reply:
I'm not a big fan of comprehensive docs essays when a simple visualization might tell you more. In this case, fully understanding the log levels for handlers and for the root is crucial.

I'd like to prevent creating multiple issues like: "Hey, I set `debug` for syslog, but it shows only `info`". Even when we have it documented, I like to have it as clear as possible. 

Yes, I've been thinking the same about maintainability, but I've convinced myself to have it there. The logic that is visualized is the standard for Quarkus logging and will not change drastically - might be extended, but not changed (AFAIK). That's why I also pinged @dmlloyd for more information. IMHO, the visualization brings more clearance on the log levels and can bring some value to users, at least until the time when sth is changed. 

I expect that the relevance will last a big portion of the time, and it can be removed when it's not relevant anymore.

@vmuzikar WDYT?

Reply:
It's not a blocker for me to keep the visualization. I just think the concrete examples below the images made it even more clear than the images itself. And that's a not a lot of text. At least I felt this way. :)

Reply:
@vmuzikar It depends on our consensus here, I'm not declining to remove it :) I just think it brings some added value as the log-level is more understandable, even for some special cases - it helps to imagine the principle better. As even Quarkus does not provide a clear explanation, I like it this way.

But if it can be found disruptive, I can remove it right away.